### ISSUE TITLE
[BUG] test.describe names don't show up in TEST EXPLORER

### ISSUE THREAD

**Issue by yogeshgadge** [microsoft/playwright#27236]
### System info
- Playwright Version: 1.38]
- Operating System: [macOS 13]
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

- [x ] I provided exact source code that allows reproducing the issue locally.

https://github.com/yogeshgadge/pw-repo/tree/how-to-combine-configs-approach1#testdescribe-is-useless-for-test-explorer

**Link to the GitHub repository with the repro**

https://github.com/yogeshgadge/pw-repo/tree/main

**Steps**
- [open project in vscode]
- [ensure Playwright for VSCode plugin is enabled]
- [Go to TEST EXPLORER]

**Expected**

Nested describe tests OR something in the flattened tests prefixing the describe text. If there are nested describe then it should have that canonicalized.

**Actual**

The description for `test.describe('my-description', () => .. ` is lost.

![image](https://github.com/microsoft/playwright/assets/2423608/ab492c6f-60c2-4644-956e-295c827c712c)

Edit:

Some not so good workaround

https://github.com/yogeshgadge/pw-repo/tree/how-to-combine-configs-approach1

**Comment by yury-s** [comment]
I cannot reproduce it, when I open VSCode for the root folder I see this:

<img width="860" alt="image" src="https://github.com/microsoft/playwright/assets/9798949/8602d382-c305-46d3-8faa-feb426cf095e">

If I open VSCode inside `projects/ui-app1/e2e` there are no tests, this is working as expected because it has config pointing to a non-existing test dir: `testDir: "./projects/ui-app1/e2e",`

Are you doing something differently?

**Comment by yogeshgadge** [comment]
My link is not good. I think you are in  a workaround branch. Can you try the main branch?

**Comment by yury-s** [comment]
> My code link is not good. I think you are in a workaround branch. Can you try the main branch?

On main branch there are no tests but this is because of the broken `testDir` filters, if I apply the following patch everything works too:

```diff
diff --git a/projects/ui-app1/e2e/playwright.config.ts b/projects/ui-app1/e2e/playwright.config.ts
index 05e4f8e..548b3fa 100644
--- a/projects/ui-app1/e2e/playwright.config.ts
+++ b/projects/ui-app1/e2e/playwright.config.ts
@@ -10,7 +10,7 @@ import { defineConfig, devices } from "@playwright/test";
  * See https://playwright.dev/docs/test-configuration.
  */
 export default defineConfig({
-  testDir: "./projects/ui-app1/e2e",
+  testDir: ".",
   /* Run tests in files in parallel */
   fullyParallel: true,
   /* Fail the build on CI if you accidentally left test.only in the source code. */
diff --git a/projects/ui-app2/e2e/playwright.config.ts b/projects/ui-app2/e2e/playwright.config.ts
index 6ee4345..548b3fa 100644
--- a/projects/ui-app2/e2e/playwright.config.ts
+++ b/projects/ui-app2/e2e/playwright.config.ts
@@ -10,7 +10,7 @@ import { defineConfig, devices } from "@playwright/test";
  * See https://playwright.dev/docs/test-configuration.
  */
 export default defineConfig({
-  testDir: "./projects/ui-app2/e2e",
+  testDir: ".",
   /* Run tests in files in parallel */
   fullyParallel: true,
   /* Fail the build on CI if you accidentally left test.only in the source code. */
```

**Comment by yury-s** [comment]
Closing per the response above, feel free to open a new issue if it doesn't work.
